### PR TITLE
Release `v10.0.0-soroban.0`

### PR DESCRIPTION
**Pull request by Shaptic**
### Breaking Changes

- Many XDR structures have changed, please refer to the `types/next.d.ts` diff for details ([#633](https://github.com/stellar/js-stellar-base/pull/633)).
- We have returned to the world in which **one** transaction contains **one** operation which contains **one** host function invocation. This means `Operation.invokeHostFunctions` is gone and `Operation.invokeHostFunction` has changed to accept `{ func, auth }`, where `func` is the correct `xdr.HostFunction` and `auth` is a list of `xdr.SorobanAuthorizationEntry` items that outline the authorization tree for the call stack ([#633](https://github.com/stellar/js-stellar-base/pull/633)). Better abstractions for creating an `xdr.HostFunction` are forthcoming, though you can still refer to `Contract.call()` for help.

### Added

- A new abstraction for dealing with large integers and `ScVal`s: see `ScInt`, `XdrLargeInt`, and `scValToBigInt` ([#620](https://github.com/stellar/js-stellar-base/pull/620)).
- A new abstraction for converting between native JavaScript types and complex `ScVal`s: see `nativeToScVal` and `scValToNative` ([#630](https://github.com/stellar/js-stellar-base/pull/630)).
- We have added two new operations related to state expiration in Soroban: `BumpFootprintExpiration` and `RestoreFootprint`. Please refer to their docstrings for details ([#633](https://github.com/stellar/js-stellar-base/pull/633)).

cc @piyalbasu @aristidesstaffieri for downstream context

### CODE REVIEW

File: src/transaction_base.js [hunk 1]
```diff
@@ -80,8 +80,9 @@ export class TransactionBase {
     );
 
     if (requiresContractSignatures) {
-      throw new Error(
-        'Soroban contract signatures are not supported in this version of the SDK.'
+      // eslint-disable-next-line no-console
```
Comment (sreuland) [inline src/transaction_base.js:83]: seems like `requiresContractSignatures` is not relevant for now, maybe just delete it and implement when concrete. this message may trigger more support to explain reasons.

**Comment by sreuland** [comment]
should [CHANGELOG](https://github.com/stellar/js-stellar-base/blob/v10.0.0-soroban.0/CHANGELOG.md) be updated to include v10 notes(this pr's description)?

File: src/transaction_base.js (hunk 1 above)
Comment (paulbellamy) [reply src/transaction_base.js:83]: No more support issues than we already have, and support is imminent anyway, so.

File: test/unit/transaction_test.js [hunk 2]
```diff
@@ -299,9 +299,7 @@ describe('Transaction', function () {
       .setTimeout(StellarBase.TimeoutInfinite)
       .build();
 
-    expect(() => tx.sign(signer)).to.throw(
-      /Soroban contract signatures are not supported in this version of the SDK./
-    );
+    tx.sign(signer);
```
Comment (paulbellamy) [inline test/unit/transaction_test.js:302]: This test seems a bit bunk now

**Review by paulbellamy — APPROVED** [review]

File: test/unit/transaction_test.js (hunk 2 above)
Comment (Shaptic) [reply test/unit/transaction_test.js]: yeah good point lol, dropped

File: src/transaction_base.js (hunk 1 above)
Comment (Shaptic) [reply src/transaction_base.js]: yeah agreed

**Pull request merged**